### PR TITLE
World and chunks

### PR DESCRIPTION
**Pull request by Runemoro**

### CODE REVIEW

**Review by shadowfacts — CHANGES_REQUESTED** [review]
Looks good, but I'm disinclined to merge this until the classes that have been moved into packages have names as well (`net/minecraft/server/world/uf` and `net/minecraft/server/world/uj`).

**Comment by Runemoro** [comment]
Done.

**Review by shadowfacts — APPROVED** [review]

File: mappings/net/minecraft/world/chunk/ReadOnlyChunk.mapping [hunk 1]
```diff
@@ -3,29 +3,52 @@ CLASS brl net/minecraft/world/chunk/ReadOnlyChunk
...
 	METHOD a addPendingBlockEntityTag (Lhs;)V
-	METHOD a setBiomes ([Lbco;)V
+	METHOD a getStructureStart (Ljava/lang/String;)Lcco;
+	METHOD a addStructureReference (Ljava/lang/String;J)V
+	METHOD a addStructureStart (Ljava/lang/String;Lcco;)V
+	METHOD a setStructureStarts (Ljava/util/Map;)V
+	METHOD a setShouldSave (Z)V
+	METHOD a setBiomeArray ([Lbco;)V
+		ARG 1 biomeArray
+	METHOD addStructureStart setStructureStart (Ljava/lang/String;Lcco;)V
```
Comment (asiekierka) [inline mappings/net/minecraft/world/chunk/ReadOnlyChunk.mapping]: Why did you just remap a Mojang-named name?

Comment (Runemoro) [reply mappings/net/minecraft/world/chunk/ReadOnlyChunk.mapping]: That's not a Mojang name, I first named it addStructureStart, then renamed it to setStructureStart, but it seems like Knit didn't find the correct entry for some reason and created a new one... I'll fix it right now.

File: mappings/net/minecraft/world/ChunkSaveHandler.mapping [hunk 2]
```diff
@@ -1,2 +1,10 @@
 CLASS brz net/minecraft/world/ChunkSaveHandler
-	METHOD b save ()V
+	METHOD a saveOneChunk ()Z
```
Comment (liach) [inline mappings/net/minecraft/world/ChunkSaveHandler.mapping]: which chunk is this saving?

**Review by liach — APPROVED** [review]

File: mappings/net/minecraft/world/ChunkSaveHandler.mapping (hunk 2 above)
Comment (Runemoro) [reply mappings/net/minecraft/world/ChunkSaveHandler.mapping]: It saves the first chunk the iterator of a HashMap containing chunks that need saving provides. All its usages are in a `while (saveOneChunk()) { ... }` loop.

Comment (asiekierka) [reply mappings/net/minecraft/world/ChunkSaveHandler.mapping]: saveNextChunk?

Comment (asiekierka) [reply mappings/net/minecraft/world/ChunkSaveHandler.mapping]: ... saveNextChunk?

Comment (Runemoro) [reply mappings/net/minecraft/world/ChunkSaveHandler.mapping]: Done

**Pull request merged**